### PR TITLE
Powers drop

### PR DESCRIPTION
**Pull request by JFranklin3000**
##### In this PR
- Added drop functionality
- Removed the drop down and hid the roll for power without activations

##### Testing
- create powers and drop on to actors
- make sure activation works and max number of powers as well.

### CODE REVIEW

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/actor/parts/items/power/container.hbs [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {{#>"systems/essence20/templates/actor/parts/misc/collapsible-item-container.hbs" items=@root.powers title='E20.ItemTypePowerPlural' dataType='power'}}
   {{#*inline "roll-button" item}}
-    <a class="rollable" data-roll-type={{item.type}} title="{{localize 'E20.PowerRollTitle'}}"><i class="fas fa-battery-bolt"></i></a>
+    {{#ifEquals item.system.canActivate true}}
+      <a class="rollable" data-roll-type={{item.type}} title="{{localize 'E20.PowerRollTitle'}}"><i class="fas fa-battery-bolt"></i></a>
+    {{else}}
+      <a class="rollable" data-roll-type={{item.type}} title="{{localize 'E20.PowerRollTitle'}}"></a>
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply templates/actor/parts/items/power/container.hbs:6]: I had it there so you could still send the description to the chat for powers that don't have an activation.

File: module/sheet-handlers/power-handler.mjs [hunk 2]
```diff
@@ -13,9 +17,41 @@ export class PowerHandler {
...
+      if (actorItem.type =='classFeature') {
+        if (power.system.type == "grid") {
+          if (actorItem.name == 'Personal Power') {
+            classFeatureId = actorItem._id;
+          }
+
+        }
+      }
+
+      if (actorItem.type == 'power') {
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-handler.mjs:34]: fixed

File: module/sheet-handlers/power-handler.mjs [hunk 3]
```diff
@@ -13,9 +17,41 @@ export class PowerHandler {
...
+
   async powerUpdate(power, dropFunc) {
+    const powerUuid = parseId(power.uuid);
+    let timesTaken  = 0;
+    let classFeatureId = "";
+
+    for (let actorItem of this._actor.items) {
+      if (actorItem.type =='classFeature') {
+        if (power.system.type == "grid") {
+          if (actorItem.name == 'Personal Power') {
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-handler.mjs:28]: fixed

File: module/sheet-handlers/power-handler.mjs [hunk 4]
```diff
@@ -13,9 +17,41 @@ export class PowerHandler {
...
+            classFeatureId = actorItem._id;
+          }
+
+        }
+      }
+
+      if (actorItem.type == 'power') {
+        if (actorItem.system.originalId == powerUuid) {
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-handler.mjs:35]: fixed

File: module/sheet-handlers/power-handler.mjs [hunk 5]
```diff
@@ -13,9 +17,41 @@ export class PowerHandler {
...
+          timesTaken++;
+          if (power.system.timesSelected == timesTaken) {
+            ui.notifications.warn(game.i18n.localize('E20.PowerAlreadyTaken'));
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-handler.mjs:38]: fixed

File: templates/actor/parts/items/power/container.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (phildominguez) [reply templates/actor/parts/items/power/container.hbs:6]: I think that's a button defined in a different spot that shows up for all items.

File: module/sheet-handlers/power-handler.mjs [hunk 6]
```diff
@@ -14,8 +18,30 @@ export class PowerHandler {
   * @param {Function} dropFunc   The function to call to complete the Power drop
   */
   async powerUpdate(power, dropFunc) {
+    const powerUuid = parseId(power.uuid);
+    let timesTaken  = 0;
```
Comment (phildominguez) [inline module/sheet-handlers/power-handler.mjs:22]:
```suggestion
    let timesTaken = 0;
```

File: templates/actor/parts/items/power/container.hbs [hunk 7]
```diff
@@ -1,17 +1,17 @@
 {{#>"systems/essence20/templates/actor/parts/misc/collapsible-item-container.hbs" items=@root.powers title='E20.ItemTypePowerPlural' dataType='power'}}
   {{#*inline "roll-button" item}}
-    <a class="rollable" data-roll-type={{item.type}} title="{{localize 'E20.PowerRollTitle'}}"><i class="fas fa-battery-bolt"></i></a>
+    {{#ifEquals item.system.canActivate true}}
+      <a class="rollable" data-roll-type={{item.type}} title="{{localize 'E20.PowerRollTitle'}}"><i class="fas fa-battery-bolt"></i></a>
```
Comment (phildominguez) [inline templates/actor/parts/items/power/container.hbs:4]: Missing localization when I click this:
image

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/actor/parts/items/power/container.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply templates/actor/parts/items/power/container.hbs:6]: ah, ok I see that now. Yep that can get pulled.

File: module/sheet-handlers/power-handler.mjs (hunk 6 above)
Comment (JFranklin3000) [reply module/sheet-handlers/power-handler.mjs:22]: fixed

File: templates/actor/parts/items/power/container.hbs (hunk 7 above)
Comment (JFranklin3000) [reply templates/actor/parts/items/power/container.hbs:4]: I was going to redo the chat card at the end but this is in the main branch. I will mark that for a future change.

**Review by phildominguez — APPROVED** [review]

**Pull request merged**